lib/widgets/expandable_text_widget.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:catchqr/config/app_ui_config.dart';
+import 'package:catchqr/theme/app_theme.dart';
+import 'package:catchqr/utils/context_extensions.dart';
+import 'package:flutter/material.dart';
+
+class ExpandableTextWidget extends StatelessWidget {
+  final int maxLinesToShow;
+  final String content;
+
+  ExpandableTextWidget({
+    super.key,
+    required this.content,
+    required this.maxLinesToShow,
+  });
+
+  final ValueNotifier<bool> _expanded = ValueNotifier(false);
+
+  @override
+  Widget build(BuildContext context) {
+
+    final TextSpan textSpan = TextSpan(
+      text: content,
+      style: context.textTheme.smallText
+    );
+
+    final TextPainter textPainter = TextPainter(
+      text: textSpan,
+      maxLines: _expanded.value ? null : maxLinesToShow,
+      textDirection: TextDirection.ltr,
+    );
+
+    textPainter.layout(maxWidth: MediaQuery.of(context).size.width);
+
+    void onSeeMoreLessTap() {
+      _expanded.value = !_expanded.value;
+    }
+
+    final int numberOfLines = textPainter.computeLineMetrics().length;
+    return ValueListenableBuilder(
```

Comment:
tbh first time I see managing state this way. Not saying it's wrong (hopefully)

Reply:
I searched a similar widget on a website. They realized state management this way. I don't know if it is proper, but I decided to trust them. If there is a need I can change it.

Reply:
Leave it for now, i think it should work :)